src/history/base.rs

```diff
@@ -89,7 +89,7 @@ pub struct SearchQuery {
     /// if given, only get results after/before this id (depending on direction)
     pub end_id: Option<HistoryItemId>,
     /// How many results to get
-    pub limit: Option<i64>,
+    pub limit: Option<usize>,
```

Comment:
Is there a particular rationale for expanding the value range here? I think `i64` was introduced to be compatible with the sqlite data types but I think we are not too specific here.

Reply:
I changed it cause the file-backed and filter used usize. I'll change it back.